.github/workflows/main.yml

```diff
@@ -5,47 +5,23 @@ on: [ pull_request, push ]
 jobs:
   build-test:
     name: Build and Test
-    runs-on: ubuntu-16.04
+    runs-on: ubuntu-latest
```

Comment:
I think we should explicitly specify version here. I suggest 20.04 for the latest LTS. Then we don't need to care about it until 2025.

Reply:
Why?

Reply:
It is a rare scenario, but I imagine if tests stop passing due to `latest` changing, we won't think of it as a possible cause until after a significant time spent on investigations.

Reply:
Using `latest` simplifies the matrix when we extend it to Windows and macOS :-)

We should also have weekly builds of master in place in the future, that would catch issues early enough.